doc/sphinx/conf.py.in

```diff
@@ -40,39 +39,59 @@ else:
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
 # ones.
 extensions = [
-    'sphinx.ext.autodoc',
-    'sphinx.ext.autosummary',
-    'sphinx.ext.mathjax',
-    'sphinx.ext.viewcode',
-    'sphinx.ext.napoleon',
-    'sphinx.ext.extlinks'
+    "sphinx.ext.autodoc",
+    "sphinx.ext.autosummary",
+    "sphinx.ext.mathjax",
+    "sphinx.ext.viewcode",
+    "sphinx.ext.napoleon",
+    "sphinx.ext.extlinks",
+    "nbsphinx",
 ]
 
+# Disable cell numbers in jupyter notebooks and show link to ipynb file
+nbsphinx_prolog = """
+{% set docname = env.doc2path(env.docname, base=False).split("/")[-1] %}
```

Comment:
This fixes the Ubuntu build error.
```suggestion
{% set docname = env.doc2path(env.docname, base=None).split("/")[-1] %}
```
Then we can revert to the old Docker image.